etcdserver/api/membership/cluster.go

```diff
@@ -315,6 +316,18 @@ func (c *RaftCluster) ValidateConfigurationChange(cc raftpb.ConfChange) error {
 					return ErrPeerURLexists
 				}
 			}
+
+			if confChangeContext.Member.IsLearner { // the new member is a learner
+				numLearners := 0
+				for _, m := range members {
+					if m.IsLearner {
+						numLearners++
+					}
+				}
+				if numLearners+1 > maxLearners {
```

Comment:
Can you help understand why we need `+1`? Or add a comment?

Reply:
The rule is that we can have at most `maxLearners` number of learners. `numLearners` counts the number of existing learners in cluster. `+1` is for the learner member we are adding right now.

Reply:
Ah, got it. LGTM>